docs/transactions.md

```diff
@@ -0,0 +1,241 @@
+`Kafka 0.11.0` includes support for `idempotent` and `transactional` capabilities in the `producer`. Idempotent delivery ensures that messages are delivered `exactly once`
+to a particular topic partition during the lifetime of a `single producer`. Transactional delivery allows producers to send data to multiple partitions such that either
+all messages are successfully delivered, or none of them are. Together, these capabilities enable `*exactly once semantics*` in Kafka.
```

Comment:
I think transactions lack some explanation here. I would add that "exactly once semantics" are for only one scenario:
- consume-process-produce under the same kafka cluster.